libraries/standard/mqtt/src/iot_mqtt_subscription.c

```diff
@@ -203,10 +272,56 @@ static bool _topicMatch( const IotLink_t * pSubscriptionLink,
     if( ( nameIndex == topicNameLength ) && ( filterIndex == topicFilterLength ) )
     {
         status = true;
+    }
+
+cleanup:
+
+    return status;
+}
+
+/*-----------------------------------------------------------*/
+
+static bool _topicMatch( const IotLink_t * pSubscriptionLink,
+                         void * pMatch )
+{
+    bool status = false;
+
+    /* Because this function is called from a container function, the given link
+     * must never be NULL. */
+    IotMqtt_Assert( pSubscriptionLink != NULL );
```

Comment:
Seems like this should be a runtime check rather than an assert then.

Reply:
Perhaps that comment should be reworded.  It means that the caller is this function is guaranteed to never pass NULL, as it's an internal function we implemented.  So the assert for not NULL is supposed to pass, unless the library is implemented incorrectly.